internal/api/core/describe.go

```diff
@@ -19,3 +31,45 @@ func SetAPIDescription(description APIDescription) {
 func GetAPIDescription() APIDescription {
 	return apiDescription
 }
+
+func GetDescriptionFromHandler(basePath string) (string, error) {
```

Comment:
Move the bulk of this low-level code to another file (more-or-less by itself) so that other people don't have to worry about AST (which are pretty cool).